File: src/main/java/alexsocol/asjlib/ItemPattern.java
package alexsocol.asjlib;

import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;

public class ItemPattern extends Item {
	
	public ItemPattern(String modid, String name, CreativeTabs tab, int stackSize) {
        this.setCreativeTab(tab);
        this.setMaxStackSize(stackSize);
        this.setTextureName(modid + ":" + name);
        this.setUnlocalizedName(name);
    }
}

File: src/main/java/alexsocol/asjlib/extendables/ItemPattern.kt
package alexsocol.asjlib.extendables

import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.Item

class ItemPattern(modid: String, name: String, tab: CreativeTabs, stackSize: Int): Item() {
	
	init {
		this.creativeTab = tab
		this.setMaxStackSize(stackSize)
		this.setTextureName("$modid:$name")
		this.unlocalizedName = name
	}
}
